src/item.h

```diff
@@ -670,6 +677,45 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
      */
     int chip_resistance( bool worst = false ) const;
 
+    /** How much damage has the item sustained? */
+    int damage() const {
+        return damage_;
+    }
+
+    /** Minimum amount of damage to an item (state of maximum repair) */
+    int min_damage() const;
+
+    /** Maximum amount of damage to an item (state before destroyed) */
+    int max_damage() const;
+
+    /**
+     * Apply damage to item constrained by @ref min_damage and @ref max_damage
+     * @param qty maximum amount by which to adjust damage (negative permissible)
+     * @param dmg type of damage which may be passed to @ref on_damage callback
+     * @return whether item should be destroyed
+     */
+    bool mod_damage( int qty, damage_type dt = DT_NULL );
```

Comment:
An intermediate step is unnecessary and would require somehow storing the `destroyed` status within the `item`.

We can't stop someone failing to check or act upon a return code. This version is reasonably fail-safe in that failing to check the return results in an effective no-op with the resulting item guaranteed to have `damage()` between `min_damage()` and `max_damage()`.

In the above more verbose suggestion the same failure results in a partly-destroyed intermediate state.